src/components/basecamp/UserForm.tsx

```diff
@@ -0,0 +1,124 @@
+import type { User } from 'auth0'
+import clx from 'classnames'
+import { Input } from '../ui/form'
+import { useForm, FormProvider } from 'react-hook-form'
+import { MUUID_VALIDATION } from './utils'
+import axios from 'axios'
+import useSWR from 'swr'
+import { Role } from 'auth0'
+import { UserRole } from '../../js/types'
+import MultiSelect from '../ui/form/MultiSelect'
+import { useEffect } from 'react'
+
+interface UserFormProps {
+  user: User
+  onClose: () => void
+}
+
+interface HtmlFormProps {
+  roles: string[]
+  conjoinedOrgAdminOrgIds: string // Form will return one large conjoined string
+}
+
+const fetcher = async (url: string): Promise<any> => (await axios.get(url)).data
+
+/*
+ * Form for updating users.
+*/
+export default function UserForm ({ user, onClose }: UserFormProps): JSX.Element {
+  const userId = user.user_id
+  if (userId == null) {
+    return <div>Can't load user. Missing user_id.</div>
+  }
+  const { data: roles, error } = useSWR<Role[]>(`/api/basecamp/userRoles?userId=${userId}`, fetcher)
+  if (error != null) {
+    return <div>{error}</div>
+  }
+
+  // React-hook-form declaration
+  const form = useForm<HtmlFormProps>({
+    mode: 'onBlur',
+    defaultValues: {
+      roles: roles?.map(r => r.name),
+      conjoinedOrgAdminOrgIds: user.user_metadata?.orgAdminOrgIds?.join(', ') ?? ''
+    }
+  })
+  const { handleSubmit, setValue, watch, formState: { isSubmitting, dirtyFields } } = form
+
+  useEffect(() => {
+    if (roles != null) setValue('roles', roles.map(r => r.name as string))
```

Comment:
Seems like the `useForm` hook sets the defaultValues on first render, and then never revisits it. However, useSWR serves up stale data on first render and refetches in the background. When the refetch returns, useForm no longer updates the values of the form leaving the roles with inaccurate data. This effect fixes that by explicitly setting the value.

Reply:
You can call react-hook-form `reset()` and passing in new values to the form. 
- https://stackoverflow.com/questions/62242657/how-to-change-react-hook-form-defaultvalue-with-useeffect
- https://react-hook-form.com/api/useform/reset/